### ISSUE TITLE
[Standards & Directives] Editorial, Pedagogical, Notebook, and Visual Design Standards for Course Authors and AI Agents

### ISSUE THREAD

**Issue by allamaprabhuani** [CEMS-Lab/autumn-school#18]
## Delivery status — 10 September 2026

The reviewed book is [live on GitHub Pages](https://cems-lab.github.io/autumn-school/book/index.html) from [source revision `3d9739c`](https://github.com/CEMS-Lab/autumn-school/commit/3d9739c736fffc104a805c8b74cd9296cab1137c). Seven HTML pages and all 12 notebook downloads match the published source; desktop/mobile book credits and math passed live checks. See `evidence/publication_20260910.json`. Fresh Colab and full classroom rehearsal retain their separate gates.

### Completed / retained

- [x] The source-authoritative workflow now enforces tutorial anatomy, source/download parity and balanced math delimiters.
- [x] The Sphinx design gate stops invalid builds; six authoring regression tests pass.
- [x] The current shared plot setup, badge/answer controls and responsive notebook reading surfaces passed local checks.

### Remaining acceptance checks

- [ ] Continue applying the standard to every new or modified asset and complete the full student-level review.
- [ ] Preserve the separate numerical, fresh-Colab, native presentation-machine and live publication gates.

**Next action:** Keep this governing standards issue open through release review and future course contributions; #17's older exercise-count wording is superseded by this issue.

**Evidence in the 10 September source edition:** `AGENTS.md`; `DESIGN_DIRECTIVES.md`; `scripts/check_design_directives.py`; `scripts/test_course_authoring.py`; `evidence/course_delivery_20260910.md`.

All original scientific requirements below remain applicable except the explicitly updated schedule/format directives. This issue stays open until its complete acceptance checks pass.

---

## Purpose & Background

This specification establishes the authoritative **textual, pedagogical, visual, and notebook design standards** for the PhAST Autumn School course repository. Any human contributor or autonomous LLM agent authoring or modifying book chapters, lecture notes, or computational notebooks **must adhere strictly to these guidelines**.

The pedagogical and design standards are modeled after premier scientific computing and deep learning educational resources:
- **[Dive into Deep Learning (D2L.ai)](https://d2l.ai/chapter_convolutional-neural-networks/index.html)**
- **[Applied Deep Learning for Physics (ADL4P / TUM-PBS)](https://tum-pbs.github.io/ADL4P/)**

---

### A. Educational Decorum & Accessible Language
- **Audience:** The text must be written in plain, clear, and encouraging educational language accessible to an **undergraduate student** in engineering, physics, or computer science.
- **Tone:** Objective, pedagogical, welcoming, and clear. Avoid overly dramatic rhetoric, colloquial conversational filler, and dry administrative compliance bureaucracy.
- **Pedagogical Flow:** Always introduce physical intuition and real-world mechanics context *before* presenting formal variational equations, discrete tensors, or code blocks.

### B. Prohibited Phrasing & Linguistic Anti-Patterns
1. **NO Negative / Failure-Obsessed Framing:**
   - ❌ *Avoid:* "Why average predictions fail", "the stationary trap", "a fatal flaw", "these are the limiting factors", "the solver fails", "failed recovery".
   -  *Use:* "Branch selection in multi-valued energy landscapes", "stationary points in non-convex potentials", "domain of applicability", "assumptions and trade-offs", "convergence criteria".
2. **NO Binary / Contrastive AI Formulas ("X, not Y" and "While A, B is not done"):**
   - ❌ *Avoid:* "A quasi-Newton method describes a nonlinear solver, while XFEM describes a crack representation", "A neural model is an adapter, not a solver", "While that is done, this is left undone".
   -  *Use:* State concepts constructively and define their modular relationships clearly without correcting an imaginary opponent.
3. **NO Compliance Checklists & Log-Dumping Quizzes:**
   - ❌ *Avoid:* Mandating rigid "Minimum result cards" or quizzing students on verifying 10-digit floating-point outputs from specific runs (e.g. "*The model has held-out MSE 0.6249999953. Which metric confirms this?*").
   -  *Use:* Meaningful engineering exercises that test conceptual understanding (e.g., Taylor series truncation error analysis, physical consequences of branch averaging, varying length scale $\ell$ to inspect damage width).

---

## 2. Notebook Structure & Anatomy (The D2L / ADL4P Blueprint)

Every tutorial notebook must follow a standardized 6-part anatomy:

1. **Header & Action Badges:**
   - Notebook Title (`# Clear, Descriptive Title`)
   - **Learning Objective:** A 1–2 sentence pedagogical summary of what the student will build and understand.
   - Action badges: `[Download Practice Notebook]`, `[Download with Worked Solutions]`, `[Environment Setup]`, and `[Open in Colab]`.
2. **Physical Motivation & Governing Equations:**
   - Concise conceptual introduction with an intuitive physical diagram or schematic.
   - Clean mathematical formulation with all symbols defined upon appearance.
3. **Step-by-Step Code Walkthrough:**
   - Code blocks interleaved with explanatory markdown.
   - Clear comments explaining tensor dimensions, coordinate conventions, boundary masks, and PyTorch autograd calls.
4. **Physical Interpretation of Results:**
   - Visual inspection of spatial fields (displacement $u$, damage $d$) and integral curves (reaction force vs. displacement).
   - Direct explanation of what the plots reveal about material physics (elastic stretching, peak load, crack localization, softening).
5. **Key Takeaways (Summary):**
   - 3–4 bullet points summarizing the core physical and numerical concepts.
6. **Student Exercises (with Expandable Hints and Worked Solutions):**
   - **Exercise 1 (Analytical / Conceptual):** Deriving a formula, proving a property, or reasoning about physical scales.
   - **Exercise 2 (Numerical / Practical):** Modifying a parameter in code, testing convergence, or evaluating trade-offs.
   - Each exercise must feature collapsible `Hint` and `Worked Solution` dropdown admonitions.

---

### A. Document Typography & Layout
- **Prose Font:** Clean system sans-serif stack (`Inter`, `-apple-system`, `BlinkMacSystemFont`, `Segoe UI`, `Roboto`, `Helvetica Neue`).
- **Code Font:** Modern monospace stack (`ui-monospace`, `SF Mono`, `Source Code Pro`, `JetBrains Mono`, `Menlo`, `Consolas`).
- **Headings:** Bold, clean, modern headers with subtle accent bars.
- **Callout Admonitions:** Clean native Sphinx/Bootstrap admonitions (`note`, `tip`, `important`) with soft backgrounds and high contrast in both dark and light modes.

### B. Matplotlib Plotting Directives
All generated plots in notebooks and book chapters must follow strict publication-grade formatting:
```python
import matplotlib.pyplot as plt

plt.rcParams.update({
    'font.family': 'sans-serif',
    'font.sans-serif': ['DejaVu Sans', 'Arial', 'Helvetica'],
    'font.size': 11,
    'axes.labelsize': 11,
    'axes.titlesize': 12,
    'xtick.labelsize': 10,
    'ytick.labelsize': 10,
    'legend.fontsize': 10,
    'figure.titlesize': 13,
    'figure.dpi': 150,
    'axes.grid': True,
    'grid.alpha': 0.3,
    'grid.linestyle': '--',
    'lines.linewidth': 1.8,
})
```
- **Color Palettes:**
  - For line plots: High-contrast palettes with distinct line styles (`-`, `--`, `-.`) for accessibility.
  - For continuous scalar fields (damage $d$, strain energy $\psi^+$): Perceptually uniform colormaps (`viridis`, `magma`, `plasma`, `coolwarm`).
  - Label colorbars clearly with the field symbol, convention ($d=0$ intact, $d=1$ broken), and physical units.
- **Axes & Labels:** Always label axes with physical quantities and units (e.g., `Applied Displacement u [mm]`, `Reaction Force F [kN]`).
- **Layout:** Always use `plt.tight_layout()` or `layout='constrained'` to prevent label clipping.

---

## 4. Course Curriculum Mapping (Six-Hour Workshop)

The six-hour course is organized into two parallel streams:
- **3 Hours of Interactive Demonstrations:** Lecturer-led walkthroughs of continuum theory, tensor mechanics, and differentiable physics.
- **3 Hours of Hands-on Lab Sessions:** Guided interactive student execution across the 4 core pillars:

| Pillar | Core Theme | Lecture Milestone | Hands-On Lab Notebook |
| :--- | :--- | :--- | :--- |
| **1. Phase-Field Fundamentals** | Physics of brittle fracture & diffuse crack bands | Chapters 1 & 2: Griffith energy, regularisation $\ell$, $g(d)$, AT1 vs. AT2 | **Lab 02:** Differentiating degradation laws (`torch.autograd` vs. finite differences) |
| **2. End-to-End Simulation** | Finite element tensors & staggered solver | Chapters 3 & 4: Weak forms, T3 meshing, boundary conditions, staggered loop | **Lab 01:** Running notched tension in PhAST, extracting $F$-$\delta$ curves & damage fields |
| **3. Differentiable Mechanics** | Automatic differentiation & inverse discovery | Chapter 5 & 5a: Computational graphs, VJPs, gradient verification | **Lab 03:** Differentiable 1D bar, sensitivity verification, modulus recovery via gradient descent |
| **4. Deep Learning Integration** | Plug-and-play neural surrogates & hybrid loops | Chapter 6: Neural operators, model checkpoints, physics residual gating | **Lab 04 & 05:** Training neural field adapters, saving/reloading, residual-gated reference solver |

*Supplementary Notebook:* **Lab 00** (Branch selection in multi-valued energy landscapes) serves as a self-study teaser or opening motivation.

**Comment by allamaprabhuani** [comment]
Issue #20 now records the voice-memo curation: three primary classroom notebooks, three lecture guides, and 21 explicit example/animation outlines. The original six notebooks remain references and the existing editable slides/media are preserved. The local notebooks pass complete 91.30/4.84/5.55-second runs; visible code cells contain at most 12 nonblank lines, while setup/plot details remain expandable. Core solver, autograd and learning operations stay visible. The current 150-minute lecture plus 150-minute practical teaching core supersedes the earlier timing allocation; 60 minutes remain flexible within the event window. See `TEACHING_SCHEDULE.md`, `source/planning/PLACEHOLDER_REGISTER.md` and `evidence/voice_memo_curation_20260910.md` in the local candidate. Publication and fresh Colab/presentation rehearsal follow review.
